source/common/upstream/default_local_address_selector.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/upstream/upstream.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Upstream {
+
+/**
+ * Default implementation of UpstreamLocalAddressSelector.
+ *
+ * See :ref:`DefaultLocalAddressSelector
+ * <envoy_v3_api_msg_config.upstream.local_address_selector.v3.DefaultLocalAddressSelector>`
+ * for a description of the behavior of this implementation.
+ */
+class DefaultUpstreamLocalAddressSelector : public UpstreamLocalAddressSelector {
+public:
+  DefaultUpstreamLocalAddressSelector(
+      std::vector<::Envoy::Upstream::UpstreamLocalAddress> upstream_local_addresses);
```

Comment:
Nit: to make move semantics clearer

```suggestion
      std::vector<::Envoy::Upstream::UpstreamLocalAddress>&& upstream_local_addresses);
```